tests/helpers/test_condition.py

```diff
@@ -2644,6 +2636,114 @@ def _load_yaml(fname, secrets=None):
             }
         },
     }
+    assert new_descriptions == expected_descriptions
+
+    # Verify the cache returns the same object
+    assert await condition.async_get_all_descriptions(hass) is new_descriptions
+
+    # Load the light integration and check a new cache object is created
+    assert await async_setup_component(hass, DOMAIN_LIGHT, {})
+    await hass.async_block_till_done()
+
+    with (
```

Comment:
we could use an annotation to avoid having to do this context manager thing repeatedly, like in https://github.com/home-assistant/core/blob/dev/tests/components/websocket_api/test_commands.py#L3796